ecp5/pack.cc

```diff
@@ -3004,6 +3004,27 @@ void Arch::assignArchInfo()
                 ci->sliceInfo.has_l6mux = true;
         } else if (ci->type == id_DP16KD) {
             ci->ramInfo.is_pdp = (int_or_default(ci->params, id("DATA_WIDTH_A"), 0) == 36);
+
+            // Output register mode (REGMODE_{A,B}). Valid options are 'NOREG' and 'OUTREG'.
+            std::string regmode_a = str_or_default(ci->params, id("REGMODE_A"), "NOREG");
+            if (!(regmode_a == "NOREG" || regmode_a == "OUTREG"))
+                NPNR_ASSERT_FALSE_STR("bad DP16KD REGMODE_A configuration '" + regmode_a + "'");
```

Comment:
As this is a user-specifiable parameter, this should be `log_error` not `NPNR_ASSERT_FALSE_STR`.